### PR TITLE
docs: add minikube instructions for test/e2e

### PR DESCRIPTION
**Pull request by rach-id**
## Overview

Adds simple instructions to run the test/e2e in minikube. 

Adding monitoring is not straightforward, as a lot of values are hardcoded into the implementation. If there is a need to have the monitoring locally, we can create a separate issue for it.

### CODE REVIEW

**Review by rootulp — DISMISSED** [review]
No blocking feedback, thanks for the PR!

1. Does this PR close https://github.com/celestiaorg/celestia-app/issues/3340? 
2. It looks like I have lingering pods every each test invocation. For example, I invoked `make test-e2e` 4 times and now I see 4 lingering pods. Do we need a follow-up issue to clean them up?

```
$ kubectl get pods --namespace test
NAME                             READY   STATUS    RESTARTS   AGE
timeout-handler-05b5c03a-n668h   1/1     Running   0          1s
timeout-handler-9237997b-k2st6   1/1     Running   0          43s
timeout-handler-da6df496-rnpc2   1/1     Running   0          16s
timeout-handler-ed96a2e8-zrl5q   1/1     Running   0          2m23s
```

3. It looks like my `.kube/config` was overwritten to use minikube. To switch back to using the Kubernetes cluster specified in [Notion](https://www.notion.so/celestiaorg/Configuring-kubeconfig-for-knuui-0848a70da653413c848ac1ba6f5e3957?pm=c) do I just copy + paste the kube config and overwrite the minikube one? Can this README clarify that a user can run the end to end tests either locally or in remote Kubernetes cluster depending on what they specify in their local `.kubeconfig`?

**Review by evan-forbes — APPROVED** [review]
noice

File: test/e2e/readme.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# End to End Testing
+# End-to-End Testing
 
-Celestia uses the [knuu](https://github.com/celestiaorg/knuu) framework to orchestrate clusters of nodes in a network for end to end testing. This relies on Docker and a kubeconfig (in `~/.kube/config`) to access the Kubernetes cluster.
+Celestia uses the [knuu](https://github.com/celestiaorg/knuu) framework to orchestrate clusters of nodes in a network for end-to-end testing. This relies on Docker and a kubeconfig (in `~/.kube/config`) to access the Kubernetes cluster.
```
[the remark this answers is not in the document]
Comment (rootulp) [reply test/e2e/readme.md:3]: It's not a typo.

**Review by rootulp — APPROVED** [review]

**Review by rootulp — COMMENTED** [review]
Thank you so much! The Known issues is a great idea and I expect it will save other contributors a lot of time

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**